baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/kafkaservice/KafkaServiceFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.kie.baaas.dfm.app.manager.kafkaservice;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.kie.baaas.dfm.app.config.DecisionFleetManagerConfig;
+import org.kie.baaas.dfm.app.manager.KafkaService;
+
+@ApplicationScoped
+public class KafkaServiceFactory {
+    private final DecisionFleetManagerConfig fmConfig;
+
+    @Inject
+    public KafkaServiceFactory(DecisionFleetManagerConfig config) {
+        this.fmConfig = config;
+    }
+
+    public KafkaService createKafkaClient() {
+        KafkaService kc = new UndefinedKafkaService();
+        if (KafkaServiceType.OPERATE_FIRST.equalValue(fmConfig.getKafkaServiceType())) {
+            kc = new OperateFirstKafkaService();
+        } else if (KafkaServiceType.MANAGED_KAFKA.equalValue(fmConfig.getKafkaServiceType())) {
+            kc = new ManagedKafkaService();
+        }
+        return kc;
```

Comment:
I think we should throw an Exception if the user has specified a value for `baaas.dfm.kafka.service.type` that we don't support. Imagine the scenario where you have a small typo e.g: `baaas.dfm.kafka.service.type=managd-kafka`...that would default the deployment to `UndefinedKafkaService` and we wouldn't know until someone started to invoke the API.

Reply:
We can still keep the current implementation ie: with UndefinedKafkaService created for invalid type. But in the LifeCycle class, we will handle this undefined case like your comment at https://github.com/kiegroup/baaas-decision-fleet-manager/pull/9#discussion_r670382045

Reply:
since we are going to stop the DecisionRequest from being processed in LIfeCycle class before a DR is persisted if there is no Kafkaservice is defined, we dont need an exception here, as the injected bean will never be called, so no exception will be thrown. 

So the fail-fast is in LifeCycle class, there we will verify the type of current Kafka Service, if it is not defined and DR has the kafka info, we send back 400 to user.

Reply:
@myeung18 I still think we should throw an Exception here if the user specifies a Kafka Service that is not recognised. 

Imagine the scenario where we want to use Managed Kafka, but accidentally have a typo and rollout a new version of the service. We'd want the service to fail to boot rather than going live with the `UndefinedKafkaService` and waiting until the point where we are handling API requests before we recognise the error.

Reply:
@robpblake 
I thought we want to support starting up our app w/o a supported kafka service?  We want to disable Kafka integration.

We throw an exception here and fail fast during the startup time for an invalid configure, that means the app starts always either with operate-first or managed kafka?

Reply:
@myeung18 What I mean is that the user has to specify one of the following as the value of the `baaas.dfm.kafka.service.type` property:

- operate-first-kafka
- managed-kafka
- undefined

Anything that isn't one of those options should cause us to throw an Exception when we boot so that the user is not surprised in any way by which integration they have.

Reply:
Ok, I got it.